src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts

```diff
@@ -132,23 +132,25 @@ export class CellToggleMoreActions {
 	private removeDuplicatedAndStartingSeparators(actions: (Action | CellActionBase)[]): void {
 		let indexesToRemove: number[] = [];
 		for (let i = 0; i < actions.length; i++) {
-			// Never should have a separator at the beginning of the list
-			if (i === 0 && actions[i] instanceof Separator) {
-				indexesToRemove.push(0);
-			}
 			// Handle multiple separators in a row
 			if (i > 0 && actions[i] instanceof Separator && actions[i - 1] instanceof Separator) {
 				indexesToRemove.push(i);
 			}
-			if (i > 0 && actions[i] instanceof Separator && actions[i + 1] instanceof Separator) {
-				indexesToRemove.push(i);
-			}
 		}
 		if (indexesToRemove.length > 0) {
 			for (let i = indexesToRemove.length - 1; i >= 0; i--) {
 				actions.splice(indexesToRemove[i], 1);
 			}
 		}
+		for (let i = 0; i < indexesToRemove.length; i++) {
```

Comment:
You don't have to loop over the entire list - just splice the list directly if there's a separator : 

```
if(actions[0] instanceof Separator) {
   actions.splice(0, 1);
}
if(actions[actions.length - 1] instanceof Separator) {
   actions.splice(actions.length - 1, 1);
}
```